recovery.go

```diff
@@ -0,0 +1,48 @@
+package handlers
+
+import (
+	"log"
+	"net/http"
+	"runtime"
+)
+
+const defaultRecoveryStackSize = 8192
+
+type recoveryHandler struct {
+	logger    *log.Logger
+	handler   http.Handler
+	stackSize int
+}
+
+// RecoveryHandler is HTTP middleware that recovers from a panic,
+// logs the panic, writes http.StatusInternalServerError, and
+// continues to the next handler.
+//
+// Example:
+//
+//  logger := log.New(os.Stdout, "prefix: ", log.LstdFlags)
+//
+//  r := mux.NewRouter()
+//  r.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+//  	panic("Unexpected error!")
+//  })
+//
+//  recoverRouter := handlers.RecoveryHandler(logger, r)
+//  http.ListenAndServe(":1123", recoverRouter)
+func RecoveryHandler(logger *log.Logger, h http.Handler) http.Handler {
```

Comment:
Not sure about this function signature - if you just want to use the default logger (as part of package `log`) you can't actually pass it in.

I'd lean towards making it `func RecoveryHandler(h http.Handler) http.Handler` and then just printing to the default logger and using `debug.PrintStack()` to provide the trace. Package users can still capture the logs (as needed) by changing the output and/or (as is standard practice) feeding stderr into their external logger of choice.